lib/bundle/commands/check.rb

```diff
@@ -14,50 +16,78 @@ def reset!
         Bundle::BrewServices.reset!
       end
 
+      def exit_on_first_error?
+        !ARGV.include?("--all")
+      end
+
+      def output_errors?
+        ARGV.include?("--verbose")
+      end
+
       def run
-        if any_taps_to_tap? ||
-           any_casks_to_install? ||
-           any_apps_to_install? ||
-           any_formulae_to_install? ||
-           any_formulae_to_start?
+        @dsl ||= Bundle::Dsl.new(Bundle.brewfile)
+
+        checks = {
+          taps_to_tap: "Taps",
+          casks_to_install: "Casks",
+          apps_to_install: "Apps",
+          formulae_to_install: "Formulae",
+        }
+        completed_checks = []
+        errors = []
+        enumerator = exit_on_first_error? ? :any? : :each
+        work_to_be_done = checks.keys.send(enumerator) do |check_method|
+          check_errors = send(check_method)
+          completed_checks << check_method
+          any_errors = check_errors.any?
+          errors.concat(check_errors) if any_errors
+          any_errors
+        end
+
+        if work_to_be_done || any_formulae_to_start?
           puts "brew bundle can't satisfy your Brewfile's dependencies."
+          if output_errors?
+            errors.each { |package| puts "#{@arrow} #{package}" }
+          end
+          unchecked_checks = (checks.keys - completed_checks)
+          unchecked_checks.each { |unchecked| puts "#{checks[unchecked]} were not checked." }
+          puts "Run `brew bundle check --all` to check all dependency categories."
           puts "Satisfy missing dependencies with `brew bundle install`."
           exit 1
         else
           puts "The Brewfile's dependencies are satisfied."
         end
       end
 
-      def any_casks_to_install?
-        @dsl ||= Bundle::Dsl.new(Bundle.brewfile)
+      def casks_to_install
         requested_casks = @dsl.entries.select { |e| e.type == :cask }.map(&:name)
-        requested_casks.any? do |c|
-          !Bundle::CaskInstaller.cask_installed_and_up_to_date?(c)
+        actionable = requested_casks.reject do |c|
```

Comment:
Probably want to pass through `enumerator` and use it here (and below) so you can still get the `any?` short circuit behaviour

Reply:
Ohh, you'd like to still see the short circuit _within each category_, not just at the category level. I think I understand now.

Reply:
I started passing the enumerator through, but it got ugly fast. The check methods would return either a boolean or an array: `any?` exists on arrays but not on booleans. I'll have to refactor the bodies of the checks to separate the errors from the return value like I did for the `work_to_be_done` loop.

Maybe tomorrow night?

Reply:
@colindean Exactly, thanks! Basically: as soon as anything is missing and we're not `--verbose`: stop checking.

Reply:
I'm getting into some major rework on this but it's making the code a lot more readable.